src/main/java/com/jmb/springfactory/service/statistic/StatisticServiceImpl.java

```diff
@@ -0,0 +1,149 @@
+import com.jmb.springfactory.dao.group.GroupMongoService;
+import com.jmb.springfactory.dao.task.TaskMySQLService;
+import com.jmb.springfactory.dao.worklog.WorkLogMySQLService;
+import com.jmb.springfactory.model.bo.QueryTaskObject;
+import com.jmb.springfactory.model.dto.StatisticDto;
+import com.jmb.springfactory.model.entity.Task;
+import com.jmb.springfactory.model.entity.WorkGroup;
+import com.jmb.springfactory.model.entity.WorkLog;
+import com.jmb.springfactory.service.BaseService;
+
+@Service
+public class StatisticServiceImpl extends BaseService implements StatisticService {
+
+    @Autowired
+    private TaskMySQLService taskMySQLService;
+
+    @Autowired
+    private WorkLogMySQLService workLogMySQLService;
+
+    @Autowired
+    private GroupMongoService groupMongoService;
+
+    @Autowired
+    private CommentMySQLService commentMySQLService;
+
+    @Override
+    public StatisticDto buildStatisticInformationByOrder(final Integer scheduleId, final Integer orderId) {
+
+        serviceLog.info(String.format("Search task in schedule %s and order %s", scheduleId, orderId));
+        final StatisticDto statistics = new StatisticDto();
+        final QueryTaskObject queryTask = QueryTaskObject.builder().orderId(orderId).scheduleId(scheduleId).build();
+        final List<Task> tasks = taskMySQLService.findAll(queryTask).collect(Collectors.toList());
+
+        if (CollectionUtils.isNotEmpty(tasks)) {
+
+            serviceLog.info("Extracting general statistic data");
+            statistics.setPriority(extractPriorityData(tasks));
+            statistics.setStatus(estractStatusData(tasks));
+            statistics.setEstimatedHours(extractEstimatedHours(tasks));
+            statistics.setWorkedHours(extractWorkedHours(tasks));
+
+            serviceLog.info("Extracting group statistic data");
+            statistics.setWorkedHoursByGroup(extractGroupTaskWorkedHours());
+            statistics.setTaskByGroup(extractGroupTask());
+
+            serviceLog.info("Extracting comment statistic data");
+            statistics.setCommentByTask(extractTaskComment(tasks));
+
+            serviceLog.info("Calculating total estimated and worked time");
+            statistics.setTotalEstimatedTime(calculateTotalTime(statistics.getEstimatedHours()));
+            statistics.setTotalWorkedHours(calculateTotalTime(statistics.getWorkedHours()));
+        }
+
+        return statistics;
+    }
+
+    private Long calculateTotalTime(Map<String, Long> estimatedHours) {
+        return estimatedHours.entrySet().stream().map(Entry::getValue).reduce(Long::sum).orElse(0L);
+    }
+
+    private Map<String, Long> extractTaskComment(final List<Task> tasks) {
+
+        final Function<Task, SimpleEntry<String, Long>> countCommentByTask = task -> {
+            final Long numberOfComments = commentMySQLService.countByTask(task.getId());
+            return new SimpleEntry<String, Long>(task.getName(), numberOfComments);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)